File: app/src/main/java/ru/merkulyevsasha/news/helpers/BroadcastHelper.java
package ru.merkulyevsasha.news.helpers;

import android.content.Context;
import android.content.Intent;

/**
 * Created by sasha_merkulev on 22.09.2017.
 */

public class BroadcastHelper {

    public static final String ACTION_NAME="ru.merkulyevsasha.news.DATA_LOADING";
    public static final String KEY_FINISH_NAME="ru.merkulyevsasha.news.key_finished";
    public static final String KEY_UPDATE_NAME="ru.merkulyevsasha.news.key_updated";

    private final Context context;

    public BroadcastHelper(Context context){
        this.context = context;
    }

    private void sendBroadcast(boolean updated, boolean finished){
        Intent intent = new Intent(ACTION_NAME);
        intent.putExtra(KEY_UPDATE_NAME, updated);
        intent.putExtra(KEY_FINISH_NAME, finished);
        context.sendBroadcast(intent);
    }

    public void sendFinishBroadcast(){
        sendBroadcast(false, true);
    }

    public void sendUpdateBroadcast(){
        sendBroadcast(true, false);
    }
}

File: app/src/main/java/ru/merkulyevsasha/news/helpers/BroadcastHelper.kt
package ru.merkulyevsasha.news.helpers

import android.content.Context
import android.content.Intent
import android.support.v4.content.LocalBroadcastManager

import javax.inject.Inject

class BroadcastHelper @Inject
constructor(private val context: Context) {

    private fun sendBroadcast(start: Boolean, finished: Boolean) {
        val intent = Intent(ACTION_LOADING)
        intent.putExtra(KEY_START_WORKER, start)
        intent.putExtra(KEY_FINISH_WORKER, finished)
        LocalBroadcastManager.getInstance(context).sendBroadcast(intent)
    }

    fun notifyWorkerFinished() {
        sendBroadcast(false, true)
    }

    fun notifyWorkerStart() {
        sendBroadcast(true, false)
    }

    companion object {

        const val ACTION_LOADING = "ru.merkulyevsasha.news.DATA_LOADING"
        const val KEY_FINISH_WORKER = "ru.merkulyevsasha.news.key_finished"
        const val KEY_START_WORKER = "ru.merkulyevsasha.news.key_start"
    }
}
